From ab43d8cf291c5a517841ac0ae445c9bc6d74cf5c Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 1 Jul 2011 14:18:09 +0200
Subject: [PATCH 02/14] carl9170 firmware: drop timer1

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 carlfw/src/main.c |    8 --------
 1 files changed, 0 insertions(+), 8 deletions(-)

--- a/carlfw/src/main.c
+++ b/carlfw/src/main.c
@@ -76,8 +76,6 @@ void clock_set(enum cpu_clock_t clock_,
 	case AHB_80_88MHZ:
 		break;
 	}
-
-	timer_init(1, (fw.ticks_per_usec * 25) >> 1);
 }
 
 static void init(void)
@@ -145,10 +143,6 @@ static void timer0_isr(void)
 #endif /* CONFIG_CARL9170FW_DEBUG_LED_HEARTBEAT */
 }
 
-static void timer1_isr(void)
-{
-}
-
 static void handle_timer(void)
 {
 	uint32_t intr;
@@ -168,8 +162,6 @@ static void handle_timer(void)
 
 	HANDLER(intr, BIT(0), timer0_isr);
 
-	HANDLER(intr, BIT(1), timer1_isr);
-
 	if (intr)
 		DBG("Unhandled Timer Event %x", (unsigned int) intr);
 
