From 4fc166c3e49e614efcfc4aeac1d84bc324730033 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sun, 10 Jul 2011 18:48:05 +0200
Subject: [PATCH 05/14] carl9170 firmware: shared register file update

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 include/shared/hw.h |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

--- a/include/shared/hw.h
+++ b/include/shared/hw.h
@@ -174,6 +174,7 @@
 #define		AR9170_MAC_SNIFFER_ENABLE_PROMISC	BIT(0)
 #define		AR9170_MAC_SNIFFER_DEFAULTS		0x02000000
 #define	AR9170_MAC_REG_ENCRYPTION		(AR9170_MAC_REG_BASE + 0x678)
+#define		AR9170_MAC_ENCRYPTION_MGMT_RX_SOFTWARE	BIT(2)
 #define		AR9170_MAC_ENCRYPTION_RX_SOFTWARE	BIT(3)
 #define		AR9170_MAC_ENCRYPTION_DEFAULTS		0x70
 
