From 67460e54bedffbc62589ab4d5742e85091107b87 Mon Sep 17 00:00:00 2001
From: Pavel Roskin <proski@gnu.org>
Date: Fri, 15 Jul 2011 23:00:49 +0000
Subject: [PATCH 08/14] carl9170 firmware: install firmware without execute
 permissions

Signed-off-by: Pavel Roskin <proski@gnu.org>
Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 autogen.sh |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

--- a/autogen.sh
+++ b/autogen.sh
@@ -32,7 +32,8 @@ case "$1" in
 			tools/src/miniboot a carlfw/carl9170.fw minifw/miniboot.fw
 		fi
 
-		sudo install carlfw/carl9170.fw /lib/firmware/carl9170-$CONFIG_CARL9170FW_RELEASE_VERSION.fw
+		sudo install -m 644 carlfw/carl9170.fw \
+			/lib/firmware/carl9170-$CONFIG_CARL9170FW_RELEASE_VERSION.fw
 		echo "done."
 	;;
 
