From e01ac8f80ae7f976c02b6dd594fef7a05fb1e214 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Mon, 21 Nov 2011 20:13:20 +0100
Subject: [PATCH 18/63] carl9170 firmware: toolchain binutils 2.22

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 toolchain/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/toolchain/Makefile
+++ b/toolchain/Makefile
@@ -1,4 +1,4 @@
-BINUTILS_VER=2.21.1
+BINUTILS_VER=2.22
 BINUTILS_URL=http://mirrors.kernel.org/gnu/binutils/binutils-$(BINUTILS_VER).tar.bz2
 BINUTILS_TAR=binutils-$(BINUTILS_VER).tar.bz2
 
