From c96aef49d34bed771817e1ea1e80163c3d971a88 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Sat, 3 Dec 2011 10:46:19 +0100
Subject: [PATCH 40/63] carl9170 firmware: fix firmware build script

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 autogen.sh |    1 +
 1 file changed, 1 insertion(+)

--- a/autogen.sh
+++ b/autogen.sh
@@ -24,6 +24,7 @@ case "$1" in
 		fi
 
 		. ./.config
+		make
 
 		echo -n "Installing firmware..."
 		if [ "$CONFIG_CARL9170FW_BUILD_TOOLS" = "y" ] &&
