From 8b944e7f0c1bd11fab1f1ce1862367f5e077bad0 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Fri, 23 Dec 2011 18:15:01 +0100
Subject: [PATCH 44/63] carl9170 firmware: get rid of nested wlan_tx calls

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 carlfw/src/wlan.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/carlfw/src/wlan.c
+++ b/carlfw/src/wlan.c
@@ -343,8 +343,6 @@ static void _wlan_tx(struct dma_desc *de
 			    get(AR9170_MAC_REG_AMPDU_FACTOR),
 			    8 << super->s.ampdu_factor));
 	}
-
-	__wlan_tx(desc);
 }
 
 /* propagate transmission status back to the driver */
@@ -498,6 +496,7 @@ void __hot wlan_tx(struct dma_desc *desc
 #endif /* CONFIG_CARL9170FW_CAB_QUEUE */
 
 	_wlan_tx(desc);
+	__wlan_tx(desc);
 	wlan_trigger(BIT(super->s.queue));
 }
 
@@ -759,6 +758,7 @@ void wlan_cab_flush_queue(const unsigned
 
 		/* ready to roll! */
 		_wlan_tx(desc);
+		__wlan_tx(desc);
 		wlan_trigger(BIT(super->s.queue));
 	}
 }
