From 6c062738b542597a35517ad1a743cab3364a07aa Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Thu, 29 Dec 2011 03:18:19 +0100
Subject: [PATCH 50/63] carl9170 firmware: toolchain newlib 1.20.0

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 toolchain/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/toolchain/Makefile
+++ b/toolchain/Makefile
@@ -2,7 +2,7 @@ BINUTILS_VER=2.22
 BINUTILS_URL=http://mirrors.kernel.org/gnu/binutils/binutils-$(BINUTILS_VER).tar.bz2
 BINUTILS_TAR=binutils-$(BINUTILS_VER).tar.bz2
 
-NEWLIB_VER=1.19.0
+NEWLIB_VER=1.20.0
 NEWLIB_URL=ftp://sources.redhat.com/pub/newlib/newlib-$(NEWLIB_VER).tar.gz
 NEWLIB_TAR=newlib-$(NEWLIB_VER).tar.gz
 
