From 529acea3f4e997a5ba0d1b3cfaa784956f65be46 Mon Sep 17 00:00:00 2001
From: Christian Lamparter <chunkeey@googlemail.com>
Date: Wed, 14 Mar 2012 20:57:11 +0100
Subject: [PATCH 51/63] carl9170 firwmare: toolchain gcc 4.6.3

Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
 toolchain/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/toolchain/Makefile
+++ b/toolchain/Makefile
@@ -6,7 +6,7 @@ NEWLIB_VER=1.20.0
 NEWLIB_URL=ftp://sources.redhat.com/pub/newlib/newlib-$(NEWLIB_VER).tar.gz
 NEWLIB_TAR=newlib-$(NEWLIB_VER).tar.gz
 
-GCC_VER=4.6.2
+GCC_VER=4.6.3
 GCC_URL=http://mirrors.kernel.org/gnu/gcc/gcc-$(GCC_VER)/gcc-core-$(GCC_VER).tar.bz2
 GCC_TAR=gcc-core-$(GCC_VER).tar.bz2
 
